plugins/panels/similarity_search/run_manager.py

```
@@ -11,11 +11,28 @@
 
+def get_head_dataset_id(dataset):
+    """Get the head dataset ID for a given dataset or a snapshot.
+
+    Args:
+        dataset: the dataset instance
+
+    Returns:
+        the head dataset ID as a string
+    """
+    dataset_id = str(dataset._doc.id)
+    return dataset_id
```

Ruff (0.15.9)

[warning] Unnecessary assignment to `dataset_id` before `return` statement

Remove unnecessary assignment

(RET504)